src/Domain/Entity/ContactMediaMap.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Domain\Entity;
+
+use BBC\ProgrammesPagesService\Domain\Enumeration\ContactMediaEnum;
+use BBC\ProgrammesPagesService\Domain\ValueObject\ContactMedia;
+use InvalidArgumentException;
+
+class ContactMediaMap
+{
+    /**
+     * Example:
+     *  [
+     *      'email'    => [<ContactMedia>, <ContactMedia>, <ContactMedia>],
+     *      'facebook' => [<ContactMedia>],
+     *  ]
+     *
+     * @var ContactMedia[][]
+     */
+    private $contactsMapByMedia = [];
+
+    /**
+     * Example: ['reddit', 'email']
+     *
+     * @return string[]
+     */
+    public function getNamesOfUsedMedia(): array
+    {
+        return array_keys($this->contactsMapByMedia);
+    }
+
+    /**
+     * @return ContactMedia[]
+     */
+    public function getContactsByMedia(string $contactMediaType): array
```

Comment:
When will we need to get all contact info grouped by the type?

In v2 I don't think we display all `email`s together for example, we just display things in the order that they were defined in iSite. Is there some corner case I'm not aware of?